torch/onnx/symbolic_opset9.py

```diff
@@ -3784,6 +3784,7 @@ def empty_like(
 def new_empty(
     g: jit_utils.GraphContext, self, sizes, dtype, layout, device, pin_memory=False
 ):
+    dtype = symbolic_helper._try_get_scalar_type(dtype)
```

Comment:
Is this correct use case for `_try_get_scalar_type`? Based on previous usage below on `self`, it seems to imply this line is trying to get the dtype of `dtype`, instead of trying to retrieve its value.

Reply:
In both cases, the `dtype` is being fetched. Then we adopt a policy in which a explicit type is given through `dtype`, the "implicit" dtype of input (`self`) is used. Makes sense?

Reply:
`self` is a `Value` representing tensor, `_try_get_scalar_type` retrieves tensor data type as scalar type.
`dtype` is a `Value` representing dtype, its **value** is the scalar type. I'm not sure `_try_get_scalar_type` is the correct function for this purpose.

Reply:
Doesn't the constant value inside the `dtype` argument (which is a `torch._C._Value`) match the enum value of the underlying `torch._C.Value.type().dtype()` type?

```python
> /opt/pytorch/torch/onnx/symbolic_opset9.py(3890)new_zeros()
-> dtype = symbolic_helper._try_get_scalar_type(dtype)
(Pdb) dtype
16 defined in (%16 : Long(device=cpu) = onnx::Constant[value={4}](), scope: test_pytorch_onnx_onnxruntime.TestONNXRuntime.test_new_zeros.<locals>.Zero_::
)
(Pdb) symbolic_helper._try_get_scalar_type(dtype)
<JitScalarType.INT64: 4>
```

In the code above, I have printed `dtype` before and after `dtype = symbolic_helper._try_get_scalar_type(dtype)`. Before, `dtype` was a `torch._C.Value` with a constant 4. After it was a `JitScalarType` with value 4. In both cases, `4` was the indication of an `INT64` because `JitScalarType` uses the [same enumeration from TorchScript implementation](https://github.com/pytorch/pytorch/blob/344defc9733a45fee8d0c4d3f5530f631e823196/c10/core/ScalarType.h).

Reply:
looks like we might also have missed a few test cases where dtype value matters?